In IIncrementalValuesProvider method Where with CancellationToken is not accessible.

**Version Used**: 
Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.4"
Include="Microsoft.CodeAnalysis.CSharp" Version="4.9.2"

**Steps to Reproduce**:

1. I followed the example from the cookbook:
https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.cookbook.md#additional-file-transformation

```csharp
using Microsoft.CodeAnalysis;

namespace CustomGenerator.Generators
{
    [Generator]
    public class AttributeGenerator : IIncrementalGenerator
    {
        public void Initialize(IncrementalGeneratorInitializationContext context)
        {
            var pipeline = context.AdditionalTextsProvider
                .Where(static (text, cancellationToken) => text.Path.EndsWith(".xml"))
                .Select(static (text, cancellationToken) =>
                {
                    var name = Path.GetFileName(text.Path);
                    var code = text.GetText(cancellationToken);
                    return (name, code);
                });
        }
    }
}
```

2. I got an error CS1593 `Delegate 'Func<AdditionalText, bool>' does not take 2 arguments`

**Diagnostic Id**:

CS1593

**Expected Behavior**:

The code should compile

**Actual Behavior**:

Error during compilation

--- a/docs/features/incremental-generators.cookbook.md
+++ b/docs/features/incremental-generators.cookbook.md
@@ -191,7 +191,7 @@ public class FileTransformGenerator : IIncrementalGenerator
     public void Initialize(IncrementalGeneratorInitializationContext context)
     {
         var pipeline = context.AdditionalTextsProvider
-            .Where(static (text, cancellationToken) => text.Path.EndsWith(".xml"))
+            .Where(static (text) => text.Path.EndsWith(".xml"))
             .Select(static (text, cancellationToken) =>
             {
                 var name = Path.GetFileName(text.Path);

